File: modules/wem-admin-ui/src/main/resources/web/admin/common/js/content/ContentId.ts
module api.content {

    export class ContentId implements api.Equitable {

        private value: string;

        constructor(value: string) {
            if (!ContentId.isValidContentId(value)) {
                throw new Error("Invalid content id: " + value)
            }
            this.value = value;
        }

        toString(): string {
            return this.value;
        }

        equals(o: api.Equitable): boolean {

            if (!api.ObjectHelper.iFrameSafeInstanceOf(o, ContentId)) {
                return false;
            }

            var other = <ContentId>o;

            if (!api.ObjectHelper.stringEquals(this.value, other.value)) {
                return false;
            }

            return true;
        }

        static isValidContentId(id: string): boolean {
            return !api.util.StringHelper.isEmpty(id) && !api.util.StringHelper.isBlank(id);
        }

        static fromReference(reference: api.util.Reference) {
            return new ContentId(reference.getNodeId());
        }
    }
}

File: modules/core-api/src/main/java/com/enonic/wem/api/content/ContentId.java
package com.enonic.wem.api.content;

import com.enonic.wem.api.support.AbstractId;
import com.enonic.wem.api.util.Reference;

public class ContentId
    extends AbstractId
{
    protected ContentId( final String id )
    {
        super( id );
    }

    public static ContentId from( String id )
    {
        return new ContentId( id );
    }

    public static ContentId from( final Reference reference )
    {
        return new ContentId( reference.getNodeId().toString() );
    }
}

